### PR TITLE
Revert "Revert "Planner prepare filters for analysis""

### PR DESCRIPTION
**Pull request by KochetovNicolai**
Reverts ClickHouse/ClickHouse#53782

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 6128146371bfa02384636f915b9b98f8a39bfe53 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53792/6128146371bfa02384636f915b9b98f8a39bfe53/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Planner/PlannerJoinTree.cpp [hunk 1]
```diff
@@ -84,6 +86,10 @@ namespace
 /// Check if current user has privileges to SELECT columns from table
 void checkAccessRights(const TableNode & table_node, const Names & column_names, const ContextPtr & query_context)
 {
+    /// StorageDummy is created on preliminary stage, igore access check for it.
```
Comment (Algunenano) [inline src/Planner/PlannerJoinTree.cpp:89]:
```suggestion
    /// StorageDummy is created on preliminary stage, ignore access check for it.
```

**Review by novikd — APPROVED** [review]
LGTM, but merged with master to check failed tests.

**Comment by devcrafter** [comment]
Something wrong happened in `test_drop_is_lock_free/test.py::test_query_is_permanent[detach-part-TxNotFinished]`

https://pastila.nl/?0014f436/83ba616b3eb369d159d3e8f299f0119e#qsR8c8RpXWFX1PrTA6N72Q==

**Comment by Algunenano** [comment]
Having a look at the PR since it's useful to implement parallel replicas decision on top of the analyzer.

* `test_query_is_lock_free[detach table]` currently ignored with the analyzer. It's a different error message and seems ok to adapt (there are similar fixes in other tests).
* `test_query_is_permanent[detach-part-TxNotFinished]`. It failed to wait for the locking query to start. The logs have been rolled out.


The query did run:
```
SELECT
    event_date,
    event_time_microseconds,
    type AS query
FROM system.query_log
WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'

Query id: 9ab6c16f-c095-4960-a3e1-9fb9911dd1cd

┌─event_date─┬────event_time_microseconds─┬─query───────┐
│ 2023-10-09 │ 2023-10-09 18:25:57.720389 │ QueryStart  │
│ 2023-10-09 │ 2023-10-09 18:26:03.736292 │ QueryFinish │
└────────────┴────────────────────────────┴─────────────┘
```


Checking query logs:
```
┌─event_date─┬────event_time_microseconds─┬─query────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ 2023-10-09 │ 2023-10-09 18:25:57.727613 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
│ 2023-10-09 │ 2023-10-09 18:25:58.435441 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
│ 2023-10-09 │ 2023-10-09 18:25:58.601061 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
│ 2023-10-09 │ 2023-10-09 18:25:59.879914 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
│ 2023-10-09 │ 2023-10-09 18:26:01.232269 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
│ 2023-10-09 │ 2023-10-09 18:26:02.645145 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
│ 2023-10-09 │ 2023-10-09 18:26:03.798331 │ SELECT count() FROM system.processes WHERE query_id = 'select-table_test_query_is_permanent_detach_part_TxNotFinished_'; │
└────────────┴────────────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

The problem seems to be that the query finished before the last check (after the query starts there are 5 checks separated by 1 second each).

The followup question would be why did the query finish if it's supposed to sleep for 3 x 50 seconds? The logs confirm it should have slept for that time: `'SleepFunctionCalls':50,'SleepFunctionMicroseconds':150000000,` but it only lasted for 6 seconds (`query_duration_ms:                     6015`) so the query did not do what's supposed to have done. In any case, it's unrelated to the changes.

**Comment by Algunenano** [comment]
Reported the sleep problem [here](https://github.com/ClickHouse/ClickHouse/issues/55667) and I'll open a different PR to enable the `test_query_is_lock_free[detach table]` test with the analyzer.

**Pull request merged**